### PR TITLE
Modified Oswald efficiency using the Hoerner method

### PR DESCRIPTION
**Pull request by stankarpuk93**

### CODE REVIEW

**Comment by stankarpuk93** [comment]
I have a question about the Travis Cl. Is it my mistake that I see the following message 'module 'numpy' has no attribute 'rank' '?

**Comment by timdmacdo** [comment]
The numpy issue is due to an update on the numpy side. We should have a fix for this soon but it’s not due to anything on your side.

File: trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Drag/induced_drag_aircraft.py [hunk 1]
```diff
@@ -52,12 +53,22 @@ def induced_drag_aircraft(state,settings,geometry):
...
     
+    if 'fuselage' in geometry.fuselages:
+        d_f = geometry.fuselages['fuselage'].width
+    else:
+        d_f = 0
+        
     if e == None:
-        e = 1/((1/wing_e)+np.pi*ar*K*CDp)
+        s     = 1 - 2 * (d_f/span)**2
+        f     = 0.0524*taper**4 - 0.15*taper**3 + 0.1659*taper**2 - 0.0706*taper + 0.0119
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Drag/induced_drag_aircraft.py]: Where does this fit come from?   Would be good to include.

Comment (stankarpuk93) [reply trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Drag/induced_drag_aircraft.py]: Sure, I have added the source in comments

**Comment by hao-siwt** [comment]
The attribute rank is deprecated since NumPy 1.9 and should be replace by the attribute ndim! After that it worked fine for me (:

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+from math import pi, cos, factorial
+import matplotlib.pylab as plt
+
+# ------------------------------------------------------------
+#  creates airfoil points from a CST file
+# ------------------------------------------------------------
+
+## @ingroup Methods-Geometry-Two_Dimensional-Cross_Section-Airfoil-CST
+
+class airfoil_from_CST:
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py]: suave classes should inherit from something. For example

`class airfoil_from_CST(Data):`

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+# Created:  Sep 2019, S.Karpuk
+# Modified: 
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+from SUAVE.Core import Data
+
+import numpy as np
+from math import pi, cos, factorial
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py]: don't use math functions. Instead rely on numpy. Math functions are generally slower and it's another import we need to track.

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+## @ingroup Methods-Geometry-Two_Dimensional-Cross_Section-Airfoil-CST
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py]: this is not a method. This is a class which should be a component attribute. See this folder:

https://github.com/suavecode/SUAVE/tree/develop/trunk/SUAVE/Components/Wings/Airfoils

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+# ------------------------------------------------------------
+#  creates airfoil points from a CST file
+# ------------------------------------------------------------
+
+## @ingroup Methods-Geometry-Two_Dimensional-Cross_Section-Airfoil-CST
+
+class airfoil_from_CST:
+
+    def __init__(self):
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py]: this should be a defaults not an __init__

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        w2     = [float(x) for x in filein.readline().split()]
+        y_lower,x = self.create_CST_airfoil(w1,-dz)
+        y_upper,x = self.create_CST_airfoil(w2,dz)
+            
+        return y_lower,y_upper,x
+
+
+    def create_CST_airfoil(self,w,dz):
+        """
+        Runs a CST algorythm to compute the airfoil coordinates
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/airfoil_from_CST.py]: algorithm

File: trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
+## @ingroup Methods-Geometry-Two_Dimensional-Cross_Section-Airfoil-CST
```
Comment (planes) [inline trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py]: See previous files comments

Comment (planes) [reply trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py]: How are these substantially different?

Comment (stankarpuk93) [reply trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py]: The CST definition is easier to work with if one considers optimization and one of the design variables is the airfoil shape

Comment (stankarpuk93) [reply trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py]: I have made minor changes to the Oswald efficiency, but the regression fails B737 (which makes sense) what should I do?

Comment (timdmacdo) [reply trunk/SUAVE/Methods/Geometry/Two_Dimensional/Cross_Section/Airfoil/CST/create_airfoil_from_CST.py]: One possible option is to allow both methods to exist and use a flag in the analysis setting that determines which is used. You would then need to check both of them.

**Comment by planes** [comment]
@stankarpuk93, hey as a heads up we just changed how we deal with efficiencies, both span and oswald. Let me know what you'd like to do with this PR.

-Emilio

**Comment by WallyMaier** [comment]
Whats the status here?  This PR is approaching 1 year old.

**Comment by stankarpuk93** [comment]
I will probably not include the scholtz’s method. OkeSe, close the request

**Pull request closed without merging**